yt_dlp/extractor/tenplay.py

```diff
@@ -113,3 +113,62 @@ def _real_extract(self, url):
             'uploader': 'Channel 10',
             'uploader_id': '2199827728001',
         }
+
+
+class TenPlaySeasonIE(TenPlayIE):
+    _VALID_URL = r'https?://(?:www\.)?10play\.com\.au/(?P<show>[^/]+)/episodes/(?P<season>[^/]+)$'
+
+    _TESTS = [{
+        'url': 'https://10play.com.au/masterchef/episodes/season-14',
+        'info_dict': {
+            'title': 'Season 14',
+            'id': 'MjMyOTIy',
+        },
+        'playlist_count': 65,
+        'params': {
+            'skip_download': True,
+        },
+    }, {
+        'url': 'https://10play.com.au/the-bold-and-the-beautiful-fast-tracked/episodes/season-2022',
+        'info_dict': {
+            'title': 'Season 2022',
+            'id': 'Mjc0OTIw',
+        },
+        'playlist_count': 256,
+        'params': {
+            'skip_download': True,
+        }
+    }]
+
+    def _real_extract(self, url):
+        show = self._match_valid_url(url).group("show")
+        season = self._match_valid_url(url).group("season")
+
+        season_info = self._download_json(f'https://10play.com.au/api/shows/{show}/episodes/{season}', f'{show}/{season}',)
+
+        carousels = season_info['content'][0]['components']
+
+        # find a carousel titled "Episodes", else default to the top one.
+        episodes_carousel = next(filter(lambda carousel: carousel['title'].lower() == 'episodes', carousels), carousels[0])
+
+        episodes = episodes_carousel['slides']
+        playlist_id = episodes_carousel['tpId']
+        playlist_title = season_info['content'][0].get('title')
+
+        load_more_url = urljoin(url, episodes_carousel['loadMoreUrl'])
+
+        while episodes_carousel['hasMore']:
+            skip_ids = [ep['id'] for ep in episodes]
+
+            episodes_carousel = self._download_json(
+                load_more_url, f'{show}/{season}', query={'skipIds[]': skip_ids},
+                note=f'Fetching episodes {len(episodes)}+')
+
+            episodes += episodes_carousel['items']
+
+        # make the first episode be the first episode, not the last episode
+        episodes = reversed(episodes)
+
+        return self.playlist_from_matches(
+            episodes, getter=lambda ep: urljoin(url, ep['cardLink']),
+            playlist_id=playlist_id, playlist_title=playlist_title)
```

Comment:
use a generator function for entries

Reply:
what do you mean by this?

replacing the use of getter with a generator? or something else

Reply:
For playlist extractors that make paginated requests, we want to return a generator function so that the entries can be extracted lazily and the minimum number of necessary requests will be made when options like `--playlist-items` are used.

I noticed you are using `reversed()`, but I think this not correct since the episodes for MasterChef are returned in ascending order by the API. The soap opera episodes are indeed in descending order, but since the behavior isn't consistent, I don't think we should reverse at all

Reply:
see `LastFMPlaylistBaseIE` for an example of this